sdk/core/azure-core/tests/test_utils.py

```diff
@@ -40,3 +40,21 @@ def test_case_insensitive_dict_initialization():
         assert d['platformUpdateDomainCount'] == d['platformupdatedomaincount'] == d['PLATFORMUPDATEDOMAINCOUNT'] == 5
         assert d['platformFaultDomainCount'] == d['platformfaultdomaincount'] == d['PLATFORMFAULTDOMAINCOUNT'] == 3
         assert d['virtualMachines'] == d['virtualmachines'] == d['VIRTUALMACHINES'] == []
+
+def test_case_insensitive_dict_cant_compare():
```

Comment:
can you add more tests for lowercase vs uppercase stuff, and also to get down what the behavior is with keys with multiple values etc? Thanks!

Reply:
Ill add in some unit tests for those cases. Didn't implement "one key, multiple values scenario" because that would mean storing the value as strs or converting them to str on access time. The question became do we want a CI Dict or a Headers class (which would make strs fine) and not knowing how everyone else is using case_insensensitive_dict()